src/main/java/fi/aalto/cs/apluscourses/intellij/actions/FilterOptionsActionGroup.java

```diff
@@ -25,17 +32,46 @@ public FilterOptionsActionGroup(MainViewModelProvider mainViewModelProvider) {
     this.mainViewModelProvider = mainViewModelProvider;
   }
 
-  // Caution: Array covariance in the return type.
   @Override
-  public @NotNull FilterOptionAction @NotNull [] getChildren(@Nullable AnActionEvent e) {
+  public AnAction @NotNull [] getChildren(@Nullable AnActionEvent e) {
+    this.filterOptionActions = getFilterOptionActions(e);
+    AnAction[] actions = new AnAction[this.filterOptionActions.length + 2];
+    System.arraycopy(this.filterOptionActions, 0, actions, 0, this.filterOptionActions.length);
+    actions[this.filterOptionActions.length] = new Separator();
+    if (e != null) {
+      Options filterOptions =
+              mainViewModelProvider.getMainViewModel(e.getProject()).getExerciseFilterOptions();
+      actions[this.filterOptionActions.length + 1] = new SelectAllFiltersAction(filterOptions);
+    }
```

Comment:
The array `actions` is created with a certain size, but the last element of this array is only set if `e != null`, otherwise it remains null. And indeed, according to the method signature individual actions may be null, but in reality you'll get errors (see `com.intellij.openapi.actionSystem.impl.ActionUpdater::getGroupChildren`). Perhaps it would be safer to construct the `actions` array with a size parameter dependant on whether `e` is null?

I'm also not sure when `e` would be null, so perhaps this is a somewhat contrived issue; anyway, I'll accept once it's change or I'm convinced that this is not an issue :>